src/scripts/clipperUI/clipper.tsx

```diff
@@ -620,9 +621,22 @@ class ClipperClass extends ComponentBase<ClipperState, {}> {
 
 		this.state.setState({ oneNoteApiResult: { status: Status.InProgress } });
 		SaveToOneNote.startClip(this.state).then((startClipPackage: StartClipPackage) => {
-			clipEvent.setCustomProperty(Log.PropertyName.Custom.CorrelationId, startClipPackage.responsePackage.request.getResponseHeader(Constants.HeaderValues.correlationId));
-			clipEvent.setCustomProperty(Log.PropertyName.Custom.AnnotationAdded, startClipPackage.annotationAdded);
-			this.state.setState({ oneNoteApiResult: { data: startClipPackage.responsePackage.parsedResponse, status: Status.Succeeded } });
+			RatingsHelper.incrementClipSuccessCount().then(() => {
+				// do nothing special on incrementClipSuccessCount resolve
+			}).then(() => {
+				// will happen regardless of success/failure of incrementClipSuccessCount
+				RatingsHelper.shouldShowRatingsPrompt(this.state).then((shouldShowRatingsPrompt) => {
```

Comment:
I think the key diff between this feature's storage dependencies and the way the rest of the Clipper currently seems to depend on storage is that this feature has a few `setValue` calls sprinkled around that 1) only happen on user interaction, and 2) the `getValue` calls need to wait for.

so for this particular code fragment, the thing keeping it async is that we need to wait for the `setValue(numClipSuccess)` to occur before doing the `getValue(numClipSuccess)` to determine if we want to show the ratings prompt. and the `setValue` only happens if the user decides to clip (and we don't fail them).

a similar need-to-wait scenario happens if the user clicks the "Not really..." button.

@VishaalK, what you're suggesting is using a smart value for any value being set in storage by this feature that is later required by a get from storage for this feature, yeah? that could work, but one issue I see off the bat with that is making sure the `setState`s happen in the correct order... we want `shouldShowRatingsPrompt` to be set appropriately _before_ `oneNoteApiResult.Status` is set to Succeeded.

I'm going to need to think a lot more about this... :)